### PR TITLE
Feature/boas 1733 feature flags

### PR DESCRIPTION
**Pull request by HadrianHughesKC**
## Describe your changes

* feat(packages/feature-flags): create a feature-flags package
* feat(packages/feature-flags): create a function for retrieving flags

## Issue ticket number and link

[BOAS-1733](https://pins-ds.atlassian.net/browse/BOAS-1733)

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

[BOAS-1733]: https://pins-ds.atlassian.net/browse/BOAS-1733?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/feature-flags/src/index.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+ * @returns {void}
+ * */
+
+/**
+ * @param {{ info: Logger, error: Logger }} logger
+ * @param {string} [endpoint]
+ * @returns {(featureFlagName: string, localPlanningAuthorityCode?: string) => Promise<boolean>}
+ * */
+export const isFeatureActive =
+	(logger, endpoint) => async (featureFlagName, localPlanningAuthorityCode) => {
```
Comment (beejjacobs) [inline packages/feature-flags/src/index.js:13]: We don't need `localPlanningAuthorityCode` this for applications, LPAs aren't involved. We may want a more generic name, and still allow checking user group?

File: packages/feature-flags/src/index.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+			return false;
+		}
+
+		if (endpoint.length <= 1) {
+			logger.error(`Endpoint value ${endpoint} is invalid.`);
+			return false;
+		}
+
+		const { AppConfigurationClient } = require('@azure/app-configuration');
+		const appConfigClient = new AppConfigurationClient(endpoint);
```
Comment (beejjacobs) [inline packages/feature-flags/src/index.js:29]: Is it worth doing this at the first function level, and renaming?

Then you can create a shared client for use within the app, e.g.

```
const isFeatureActive = newFeaturesClient(logger, endpoint);

// ... elsewhere
isFeatureActive ('flag-1');
isFeatureActive ('flag-2');
```

File: packages/feature-flags/src/index.js
Comment (beejjacobs) [inline packages/feature-flags/src/index.js:1]: Can we reduce info logs to debug? We have the same on appeals FO and it creates lots of log noise.

File: packages/feature-flags/src/index.js (hunk 1 above)
Comment (HadrianHughesKC) [reply packages/feature-flags/src/index.js:13]: Ah right ok, sure I'll rename and leave it as an optional argument.

**Review by beejjacobs — APPROVED** [review]
LGTM!

**Pull request merged**